docs/docs/porting-a-jekyll-website-to-gatsby.md

```diff
@@ -0,0 +1,276 @@
+}
+```
+
+### Migrating blog posts
+
+Now its time to transfer all the blog posts in to `src/content` folder.
+
+```
+  src
+  └── content
+      ├── 2007-10-29-why-every-programmer-should-play-nethack.md
+      └── 2009-04-26-barcamp-boston-4-roundup.md
+```
+
+We need to tell gatsby where to find blog posts.
+
+```js
+// gatsby-config.js
+{
+  resolve: 'gatsby-source-filesystem',
+  options: {
+    name: 'blog',
+    path: `${__dirname}/blog/`,
+  },
+},
+```
+
+In Jekyll when we create a new blog post we create something like that `YYYY-MM-DD-some-title.md` so we need to create that `slug` from the file name. The way we can do is using the `onCreateNode` method
+
+```js
+exports.onCreateNode = ({ node, getNode, actions }) => {
+  const { createNodeField } = actions
+
+  if (node.internal.type === `MarkdownRemark`) {
+    const filename = createFilePath({ node, getNode, basePath: `blog` })
+
+    // Blog files must have format name YYYY-MM-DD-title.md
+    if (node.frontmatter.layout === "post") {
+      const match = filename.match(/^\/([\d]{4}-[\d]{2}-[\d]{2})-{1}(.+)\/$/)
+      if (match) {
+        const [, date, title] = match
+        if (!date || !title) {
+          console.error(
+            `Invalid filename ${filename}. Change name to start with a valid date and title`
+          )
+        } else {
+          const slug = `blog/${slugify(date, "/")}/${title}/`
+          createNodeField({
+            node,
+            name: `slug`,
+            value: slug,
+          })
+        }
+      }
+    }
+  }
+}
+```
+
+Now we need to tell gatsby to create a page for each blog post `node`. We can do it extending a bit previous `createPages` method
+
+```js
+exports.createPages = ({ graphql, actions }) => {
+  const { createPage } = actions
+  const blogPostTemplate = path.resolve(`src/templates/blog-post.js`)
+  const pageTemplate = path.resolve(`src/templates/
+  page.js`)
+
+  return graphql(`...`).then(result => {
+    if (result.errors) {
+      return Promise.reject(result.errors)
+    }
+
+    const markdownItems = result.data.allMarkdownRemark.edges
+
+    // Create pages and blog post pages
+    markdownItems.forEach(({ node }) => {
+      if (node.frontmatter.layout === "page") {
+        createPage({
+          path: node.frontmatter.path,
+          component: pageTemplate,
+        })
+      } else if (node.frontmatter.layout === "post") {
+        createPage({
+          path: node.fields.slug,
+          component: blogPostTemplate,
+          context: {
+            slug: node.fields.slug,
+          },
+        })
+      }
+    })
+  })
+}
+```
+
+### Pagination with the blog posts
+
+What we desire is a blog section that lists all posts, really a paginated list of posts, and then let user select one to be read. The previous section creates a page for each post so we need to create the pages that list the posts. Again, we only need to update a bit the `createPages` method to create a new page for each set of six posts
```

Comment:
```suggestion
Once you've created all your blog posts, you'll want a page that lists them. This can be accomplished by updating the `createPages` method to create a new page for each set of posts.

> Note that this example includes pagination.
```